contracts/axelarnet-gateway/src/events.rs

```diff
@@ -15,13 +17,26 @@ pub enum AxelarnetGatewayEvent {
     },
 }
 
+fn add_token_attribute_if_some(token: Option<Coin>) -> impl Fn(Event) -> Event {
+    move |event| {
+        if let Some(token) = &token {
+            event.add_attribute("token", token.to_string())
+        } else {
+            event
+        }
+    }
+}
```

Comment:
generalize this as a trait on the Event? I think that can be quite useful in more cases, i.e.
```
fn add_attribute_if_some<T>(label: &str, value: Option<T>){...}
```
(or however the label is defined)

Reply:
That's an interesting thought. Let me try.